static/app/views/monitors/monitors.tsx

```diff
@@ -90,12 +90,26 @@ export default function Monitors({location}: RouteComponentProps<{}, {}>) {
       key={`${monitor.slug}-${monitorEnv?.name ?? 'no-env'}`}
       monitor={monitor}
       monitorEnv={monitorEnv}
-      onDelete={() => {
-        setApiQueryData(
-          queryClient,
-          monitorListQueryKey,
-          monitorList?.filter(m => m.slug !== monitor.slug)
-        );
+      onDelete={deletedEnv => {
+        if (deletedEnv) {
+          if (!monitorList) {
+            return;
+          }
+          const deletedEnvMonitor = monitorList.find(m => m.slug === monitor.slug);
+          if (!deletedEnvMonitor) {
+            return;
+          }
+          deletedEnvMonitor.environments = deletedEnvMonitor.environments.filter(
+            e => e.name !== deletedEnv.name
+          );
+          setApiQueryData(queryClient, monitorListQueryKey, monitorList);
```

Comment:
not sure what the best way is to optimistically update the deeper nested environment. Right now I'm
1. Finding the monitor that had its environment deleted
2. Setting its environment list to exclude the deleted environment
3. Update api query data with the same list, as the objects inside have changed

Doesn't feel super clean since i'm reusing a lot of the returned state before, but maybe its okay?

I think another option would be to just do a refetch() instead of optimistically updating, but I feel like thats a worse experience.